azure-functions/acbs-function/src/functions/acbs-amend-facility-loan-record.js

```diff
@@ -32,57 +32,57 @@ df.app.orchestration('acbs-amend-facility-loan-record', function* amendFacilityL
   const payload = context.df.input;
 
   try {
-    if (payload) {
-      const { facilityId, facility, amendments, fmr } = payload;
-      const { amendment } = amendments;
-      let facilityLoanRecordAmendments;
+    if (!payload) {
+      throw new Error('Invalid payload provided');
+    }
 
-      // 1.1. Facility Loan Record (FLR) amendment mapping
-      const flrMApped = mappings.facility.facilityLoanAmend(amendments, facility, fmr);
+    const { facilityId, facility, amendments, fmr } = payload;
+    const { amendment } = amendments;
+    let facilityLoanRecordAmendments;
 
-      // 1.2. Extract loan id for facility id
-      const loanId = yield context.df.callActivityWithRetry('get-facility-loan-id', retryOptions, {
-        facilityId,
-      });
+    // 1.1. Facility Loan Record (FLR) amendment mapping
+    const flrMApped = mappings.facility.facilityLoanAmend(amendments, facility, fmr);
```

Comment:
`Mapped`? Whilst making changes

Reply:
This is legacy code I've only removed indentation :) from what I understand it's taking a request and mapping it into something useful for the df